src/client/core/page-unload-barrier.js

```diff
@@ -0,0 +1,130 @@
+import hammerhead from './deps/hammerhead';
+import * as eventUtils from './utils/event';
+import { EventEmitter } from './utils/service';
+import delay from './utils/delay';
+
+var Promise      = hammerhead.Promise;
+var browserUtils = hammerhead.utils.browser;
+
+
+const DEFAULT_BARRIER_TIMEOUT       = 500;
+const WAIT_FOR_UNLOAD_TIMEOUT       = 3000;
+const SHORT_WAIT_FOR_UNLOAD_TIMEOUT = 30;
+
+const WAITING_FINISHED_EVENT = 'waiting-finished';
```

Comment:
I don't think so. It's an internal event for this module. In this case it will look like: pageUnloadMonitor.emit('testcafe|page-unload-monitor|waiting-finished') - it's unnecessary duplication.
I suppose we should use this approach ('testcafe|<smth>') for public consts and events